src/components/NodeToolbar/NodeToolbarFeatureLoader.js

```diff
@@ -21,14 +21,14 @@ type Props = {
   categories: CategoryLookupTables,
   hidden: boolean,
   modalNodeState: ModalNodeState,
-  isReportMode: boolean,
   onOpenReportMode: ?() => void,
   onStartPhotoUploadFlow: () => void,
   history: RouterHistory,
   onClose?: ?() => void,
   onClickCurrentMarkerIcon?: (feature: Feature) => void,
   onSelectWheelchairAccessibility?: (newValue: YesNoLimitedUnknown) => void,
   onEquipmentSelected: (placeInfoId: string, equipmentInfo: EquipmentInfo) => void,
+  onShowPlaceDetails: (featureId: string | number) => void,
```

Comment:
The `NodeToolbarFeatureLoader` is not part of this PR, I know. I still have the feeling this helper component can be greatly simplified. By creating the promise outside the component and passing it via props you can remove alot of code checking if the passed promise value is the same or not. I just created a similar example for the async header. Come by if you would like to further discuss this. But as I said: not part of this PR.

Reply:
If there is some reusable code for the AsyncHeader, feel free to create the PR :)

Reply:
0 for generating the promise inside the component. But as we said, this should be something we can use as a common component.